### DISCUSSION TITLE
Can I setting the default WYSIWYG toolbar items in every new WYSIWYG field?

### DISCUSSION THREAD

**Discussion by seanmars** [directus/directus#15812] [Archive]
As the title, Can I do this? Or How to do it?

**Comment by azrikahar** [comment]
There's no out of the box global/settings config for setting default interface options as of now, but one potential alternative would be to use a Filter hook extension or a Filter hook Flows to override the toolbar items on create.

Here's a quick example:

1. Add the environment variable `FLOWS_EXEC_ALLOWED_MODULES=lodash`. This is optional, as it's to use lodash's `merge` to simplify the merging of the default value. You can omit it then modify the code accordingly to assign the defaults.

2. Create a Filter hook Flow with scope of `items.create` on the collection `directus_fields`.

3. Add a Run Script operation with the following code:

    Note: the following IF statements can be combined as an OR condition, but for the purpose of explaining their purpose clearer, the following code opts to split them into 2 separate IF statements.
    
    ```js
    const lodash = require('lodash');
    
    module.exports = async function (data) {
        // Return early if it's not a WYSIWYG interface
        if (data.$trigger.payload?.meta?.interface !== 'input-rich-text-html') {
            return data.$trigger.payload;
        }
        // Return early if the default toolbar items was already modified on create
        if (data.$trigger.payload?.meta?.options?.toolbar !== undefined) {
            return data.$trigger.payload;
        }
    
        // Add/edit your default toolbar items here
        const defaultToolbarItems = ["h1", "h2", "h3"];
    
        return lodash.merge(
            {},
            data.$trigger.payload,
            {
                meta: {
                    options: { toolbar: defaultToolbarItems }
                }
            }
        );
    }
    ```

4. Save. It should now add your `defaultToolbarItems` value to every new WYSIWYG fields.

Quick screen capture as demo:

https://user-images.githubusercontent.com/42867097/193215339-0b7a45d8-1681-46d3-9fbd-06c911bdc29f.mp4

**Reply by seanmars** [reply]
It's really good. Thanks for the answer!

**Reply by birdream** [reply]
really helpful, thanks!
